### PR TITLE
feat: Add blueprint for defining custom rules

### PR DESCRIPTION
**Pull request by blaine-arcjet**
This is a blueprint for defining custom local rules that will run in the Arcjet SDK rules engine.

### CODE REVIEW

File: src/content/docs/blueprints/defining-custom-rules.mdx [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+title: "Defining custom rules"
+description: "Define custom local rules for the Arcjet SDK."
+---
+
+The Arcjet SDK can be augmented with custom rules written by users, as long as
+they match the format defined by our `@arcjet/protocol` package. These rules
```
Comment (davidmytton) [inline src/content/docs/blueprints/defining-custom-rules.mdx:7]:
```suggestion
they match the format defined by our [`@arcjet/protocol` package](https://github.com/arcjet/arcjet-js/tree/main/protocol). These rules
```

File: src/content/docs/blueprints/defining-custom-rules.mdx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+For example, you may want to [validate form
+input](https://blog.arcjet.com/next-js-security-checklist/#2-data-validation-and-sanitization)
+as part of your Arcjet protections before a request reaches your route handler,
+such as via Next.js middleware.
+
+## Local rule: Zod + Body validation
+
+If we want to run Zod schema validation against the request body, we need to
```
Comment (davidmytton) [inline src/content/docs/blueprints/defining-custom-rules.mdx:36]:
```suggestion
If we want to run [Zod](https://zod.dev/) schema validation against the request body, we need to
```

File: src/content/docs/blueprints/defining-custom-rules.mdx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+create a new local rule:
+
+```ts
```
Comment (davidmytton) [inline src/content/docs/blueprints/defining-custom-rules.mdx:39]: Can this go in a snippets file?

Comment (blaine-arcjet) [reply src/content/docs/blueprints/defining-custom-rules.mdx]: Done

**Review by davidmytton — APPROVED** [review]

**Pull request merged**